backend/src/VFP/Translation/TranslateToHaskellSource.hs

```diff
@@ -54,3 +55,18 @@ translateTypedToHaskell (TypedReference typ name args) =
 translateTypedToHaskell (TypedLiteral typ name) =
     "TypedLiteral " ++ brace (printHaskellType typ) ++ space ++ quote name
 
+translateToHaskellCode :: TypedValue -> String
+translateToHaskellCode (TypedValueDefinition typ name inner) = name ++ " :: " ++ (translateTypeToHaskellType typ) ++ "\n" ++ name ++ " = " ++ (translateToHaskellCode inner)
+translateToHaskellCode (TypedTypeHole _ _) = error "No type holes allowed"
+translateToHaskellCode (TypedLambda _ (_, pname) body) = brace $ "\\" ++ pname ++ " -> " ++ translateToHaskellCode body
+translateToHaskellCode (TypedReference _ "if" [condition, thenBody, elseBody]) = "if " ++ translateToHaskellCode condition ++ "\n\tthen " ++ translateToHaskellCode thenBody ++ "\n\telse " ++ translateToHaskellCode elseBody
+translateToHaskellCode (TypedReference _ name args) = name ++ " " ++ intercalate " " (map translateToHaskellCode args)
+translateToHaskellCode (TypedLiteral typ name) = case typ of
+    (Primitive _) -> name
+    _ -> error $ "Unsuppported literal type " ++ show typ
+
+translateTypeToHaskellType :: Type -> String
+translateTypeToHaskellType (Primitive n) = n
+translateTypeToHaskellType (Generic num) = [chr (ord 'a' - 1 + num)]
+translateTypeToHaskellType (List inner) = "[" ++ translateTypeToHaskellType inner ++ "]"
+translateTypeToHaskellType (Function from to) = translateTypeToHaskellType from ++ " -> " ++ translateTypeToHaskellType to
```

Comment:
This needs to be nested, you can look into the UIModel how to do it.

Example: A function, that takes a function as first argument